docs: update GitHub Pages documentation

DoodlesEpic opened the pull request:
### Description

The previous documentation would suggest to use a bash script which would either be run locally or in a third-party CI, however it's now possible to do the build and deployment step directly through a GitHub Actions workflow, the documentation has been updated to reflect that.

### Additional context

Closes #10938

---

### What is the purpose of this pull request?

- [ ] Bug fix
- [ ] New Feature
- [X] Documentation update
- [ ] Other

### Before submitting the PR, please make sure you do the following

- [X] Read the [Contributing Guidelines](https://github.com/vitejs/vite/blob/main/CONTRIBUTING.md).
- [X] Read the [Pull Request Guidelines](https://github.com/vitejs/vite/blob/main/CONTRIBUTING.md#pull-request-guidelines) and follow the [PR Title Convention](https://github.com/vitejs/vite/blob/main/.github/commit-convention.md).
- [X] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [X] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [X] Ideally, include relevant tests that fail without this PR but pass with it.

DoodlesEpic committed:
docs: update GitHub Pages documentation

The previous documentation would suggest to use a bash script which would either be run locally or in a third-party CI, however it's now possible to do the build and deployment step directly through a GitHub Actions workflow, the documentation has been updated to reflect that.

DoodlesEpic committed:
docs: fix prettier lint in static-deploy.md

For some reason the lint stage fails if there are double quotation marks inside the GitHub Actions workflow, hopefully this is the only needed change.

sapphi-red commented:
Using actions to deploy GitHub Pages is still in public beta. But given that it's been almost half a year since public beta started, maybe we can update the docs.
https://github.blog/changelog/2022-07-27-github-pages-custom-github-actions-workflows-beta/

patak-cat reviewed (dismissed):
Agreed, GitHub isn't going to break the beta API after such a long time.

sapphi-red commented on docs/guide/static-deploy.md:
```diff
@@ -62,42 +62,57 @@ Now the `preview` command will launch the server at `http://localhost:8080`.
-   # git push -f git@github.com:<USERNAME>/<REPO>.git main:gh-pages
-
-   cd -
+2. Go to your GitHub Pages configuration and choose the source of deployment as "GitHub Actions", this will lead you to create a workflow that builds and deploys your project, a sample workflow that installs dependencies and builds using Yarn is provided:
```
> - You are using npm. You can use equivalent commands to run the scripts if you are using Yarn or other package managers.

We have this sentence above. So I think we should use npm here as well.

sapphi-red commented on docs/guide/static-deploy.md:
```diff
@@ -62,42 +62,57 @@ Now the `preview` command will launch the server at `http://localhost:8080`.
+       runs-on: ubuntu-latest
+       steps:
+         - name: Checkout
+           uses: actions/checkout@v3
```
I guess we need `actions/setup-node@v3` before building.

sapphi-red commented on docs/guide/static-deploy.md:
````diff
@@ -62,42 +62,57 @@ Now the `preview` command will launch the server at `http://localhost:8080`.
    ```
 
 ::: tip
-You can also run the above script in your CI setup to enable automatic deployment on each push.
+You may have to modify the the "path" in the artifact step or the command in the build step to make sure it works with your setup.
 :::
````
I think we can remove this part as we have the following sentence above.

> - You are using the default build output location (`dist`).

DoodlesEpic commented on docs/guide/static-deploy.md:
```diff
@@ -62,42 +62,57 @@ Now the `preview` command will launch the server at `http://localhost:8080`.
+   on:
+     # Runs on pushes targeting the default branch
+     push:
+       branches: ['master']
```
Correct, I based this workflow in one of the default configurations on GitHub, but it seems it changes the workflow based on your main branch, mine was still 'master' because it was created before this change was incorporated.

DoodlesEpic committed:
docs: change 'master' to 'main' in GH Pages

'main' is now the default branch name in GitHub.

Co-authored-by: 翠 / green <green@sapphi.red>

DoodlesEpic committed:
docs: fix weird formatting in static-deploy.md

DoodlesEpic committed:
docs: use npm instead of yarn in GH Pages CI

As noted by sapphi-red in the PR at:
https://github.com/vitejs/vite/pull/12035

The rest of the documentation in this page assumes the developer is
using NPM, and that makes sense since it's the default, so this workflow
example should also default to NPM.

DoodlesEpic committed:
docs: remove unnecessary tip for GH Pages CI

The documentation at static-deploy.md assumes the developer is using the
default build path of 'dist', telling that they can change that path or
might have to do it to adjust to their setup is just stating the
obvious.

DoodlesEpic committed:
docs: use actions/setup-node in GH Pages CI

The action at https://github.com/actions/setup-node allows you to
specify the desired nodejs version which is good for consistency as the
system version might change at any moment a new Ubuntu release comes out
when using the ubuntu-latest environment.

DoodlesEpic commented:
Hey, I applied a formatting change in [0d4bfbe](https://github.com/vitejs/vite/pull/12035/commits/0d4bfbe0f18ae229aa56dfdf4b2b8221a9aeff7b) but I'm not sure it was correct to do so? Some code samples in the same file do it with this amount of padding and others do it with 3 spaces of padding, which one would be the correct way?

Otherwise, I believe I've applied all the suggestions by @sapphi-red

sapphi-red commented:
> Hey, I applied a formatting change in [0d4bfbe](https://github.com/vitejs/vite/pull/12035/commits/0d4bfbe0f18ae229aa56dfdf4b2b8221a9aeff7b) but I'm not sure it was correct to do so? Some code samples in the same file do it with this amount of padding and others do it with 3 spaces of padding, which one would be the correct way?

If the code block is a part of the list, it would need the indent. In this case, I think this code block is part of the "2. Go to your GitHub Pages...".

sapphi-red commented on docs/guide/static-deploy.md:
```diff
@@ -62,43 +62,60 @@ Now the `preview` command will launch the server at `http://localhost:8080`.
-::: tip
-You can also run the above script in your CI setup to enable automatic deployment on each push.
-:::
+2. Go to your GitHub Pages configuration and choose the source of deployment as "GitHub Actions", this will lead you to create a workflow that builds and deploys your project, a sample workflow that installs dependencies and builds using NPM is provided:
```
```suggestion
2. Go to your GitHub Pages configuration and choose the source of deployment as "GitHub Actions", this will lead you to create a workflow that builds and deploys your project, a sample workflow that installs dependencies and builds using npm is provided:
```
npm uses npm and not NPM.
https://docs.npmjs.com/about-npm

sapphi-red committed:
chore: npm

bluwy commented on docs/guide/static-deploy.md:
```diff
@@ -62,43 +62,60 @@ Now the `preview` command will launch the server at `http://localhost:8080`.
-::: tip
-You can also run the above script in your CI setup to enable automatic deployment on each push.
-:::
+2. Go to your GitHub Pages configuration and choose the source of deployment as "GitHub Actions", this will lead you to create a workflow that builds and deploys your project, a sample workflow that installs dependencies and builds using npm is provided:
```
```suggestion
2. Go to your GitHub Pages configuration in the repository settings page and choose the source of deployment as "GitHub Actions", this will lead you to create a workflow that builds and deploys your project, a sample workflow that installs dependencies and builds using npm is provided:
```

Maybe we can show where to find the gh pages config page

DoodlesEpic committed:
docs: instructions to find settings for GH Pages CI

Co-authored-by: Bjorn Lu <bjornlu.dev@gmail.com>

DoodlesEpic committed:
docs: fix indentation in GH Pages CI

sapphi-red commented on docs/guide/static-deploy.md:
```diff
@@ -62,44 +62,61 @@ Now the `preview` command will launch the server at `http://localhost:8080`.
+         - name: Set up Node
+           uses: actions/setup-node@v3
+           with:
+             node-version: 18
```
```suggestion
             node-version: 18
             cache: 'npm'
```
Let's use the cache.

sapphi-red committed:
docs: use cache option

sapphi-red reviewed (approved):
Thanks!
